src/logger.ts

```diff
@@ -41,23 +41,29 @@ export const debug_info = {
 export async function saveInterfaceStat(): Promise<void> {
   console.log(apiPerfLogData)
   try {
-    // eslint-disable-next-line prefer-const
-    let { api_name, tfinal, timestamp, nodeUrl, success, reason, hash } = apiPerfLogData[0]
-    // nodeUrl = nodeUrl ? nodeUrl : new URL(nodeUrl as string).hostname
-    let placeholders = `NULL, '${api_name}', '${tfinal}','${timestamp}', '${nodeUrl}', '${success}', '${reason}', '${hash}'`
-    let sql = 'INSERT INTO interface_stats VALUES (' + placeholders + ')'
-    for (let i = 1; i < apiPerfLogData.length; i++) {
-      // eslint-disable-next-line prefer-const
-      let { api_name, tfinal, timestamp, nodeUrl, success, reason, hash } = apiPerfLogData[i] // eslint-disable-line security/detect-object-injection
-
-      // nodeUrl = nodeUrl ? nodeUrl : new URL(nodeUrl as string).hostname
-      placeholders = `NULL, '${api_name}', '${tfinal}','${timestamp}', '${nodeUrl}', '${success}', '${reason}', '${hash}'`
-      sql = sql + `, (${placeholders})`
-    }
+    const stmt = db.prepare(`
+      INSERT INTO interface_stats 
+      ("api_name", "tfinal", "timestamp", "nodeUrl", "success", "reason", "hash") 
+      VALUES (?, ?, ?, ?, ?, ?, ?)
+    `)
+
+    const insertMany = db.transaction((items) => {
+      for (const item of items) {
+        stmt.run(
+          item.api_name,
+          item.tfinal,
+          item.timestamp,
+          item.nodeUrl,
+          item.success ? 1 : 0,
+          item.reason || null,
+          item.hash || null
+        )
+      }
+    })
 
-    await db.exec(sql)
+    insertMany(apiPerfLogData)
   } catch (e) {
-    console.log(e)
+    console.error('Error saving interface stats:', e)
```

Comment:
consider adding more useful info to the error message

Reply:
more detailed message should come from batchInsert and more general in the functions invoking batchInsert, `saveInterfaceStat` and `txStatusSaver`